### PR TITLE
Add GitHub Action for production deploy

### PR DESCRIPTION
**Pull request by mcbouslog**
Adds GitHub Action for production deploy, based on Notes from Nature Field Book GitHub Action for production deploy.

### CODE REVIEW

**Comment by mcbouslog** [comment]
I'll update README in separate, subsequent PR.

File: .github/workflows/deploy_production.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    - uses: actions/checkout@v2
+    - uses: azure/login@v1
+      with:
+          creds: ${{ secrets.AZURE_STATIC_SITES }}
+
+    - name: Node.js build
+      uses: actions/setup-node@v1
+      with:
+        node-version: '14.x'
+    - run: npm install
```
Comment (eatyourgreens) [inline .github/workflows/deploy_production.yml:26]: Not a blocking issue but `npm ci` would be more secure for deploys here. 
https://docs.npmjs.com/cli/v7/commands/npm-ci

Comment (mcbouslog) [reply .github/workflows/deploy_production.yml:26]: This repo doesn't even have a `package-lock.json` yet 😞 . Per the linked docs it notes "The project must have an existing package-lock.json or npm-shrinkwrap.json." There's discussion about refactoring this repo so at minimum it doesn't use `haw`, and certainly should use `package-lock.json`, but I think it's ok to pass on those larger refactors for now.

Comment (eatyourgreens) [reply .github/workflows/deploy_production.yml:26]: D’oh! I missed that. Yeah, you can’t freeze the dependencies if there’s no lock file.

Comment (mcbouslog) [reply .github/workflows/deploy_production.yml:26]: I really really should (will!) add the lockfile asap, regardless of larger refactors. I'm scared to touch more than one thing at a time in this repo though, so doing things kinda crazily/overly small and incrementally.

**Review by eatyourgreens — APPROVED** [review]
LGTM!

**Pull request merged**